discovery/marathon/marathon.go

```diff
@@ -98,6 +100,7 @@ func NewDiscovery(conf *config.MarathonSDConfig) (*Discovery, error) {
 		servers:         conf.Servers,
 		refreshInterval: time.Duration(conf.RefreshInterval),
 		appsClient:      fetchApps,
+		token:           os.Getenv(conf.TokenEnvName),
```

Comment:
We don't go via env vars in this case. The token has to go into the config file, and is then hidden in the output if it matches `patAuthLine`, see https://github.com/prometheus/prometheus/blob/master/config/config.go#L33 .